### PR TITLE
Merging ProjectManagement into PackageManagement

### PR DESCRIPTION
**Pull request by emgarten**
* Merging ProjectManagement into PackageManagement
* Adding csproj references to resource strings which were missed by auto migration.
* Cleaning up resource strings.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Strings.Designer.cs [hunk 1]
```diff
@@ -8,34 +8,35 @@
 // </auto-generated>
 //------------------------------------------------------------------------------
 
-namespace NuGet.Build.Tasks.Pack {
+namespace NuGet.Build.Tasks.Pack.Library {
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Strings.Designer.cs]: All other classes are in `NuGet.Build.Tasks.Pack` namespace. Any reason for this change? It pollutes usings with non-obvious  entries :(

File: src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs [hunk 2]
```diff
@@ -1477,7 +1477,7 @@ await PreviewBuildIntegratedProjectActionsAsync(buildIntegratedProject, actions,
             var oldListOfInstalledPackages = projectInstalledPackageReferences.Select(p => p.PackageIdentity);
             if (oldListOfInstalledPackages.Any(p => p.Equals(packageIdentity)))
             {
-                var alreadyInstalledMessage = string.Format(ProjectManagement.Strings.PackageAlreadyExistsInProject, packageIdentity, projectName ?? string.Empty);
+                var alreadyInstalledMessage = string.Format("Package '{0}' already exists in project '{1}'", packageIdentity, projectName ?? string.Empty);
```
Comment (alpaix) [inline src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs]: Move it to `Strings`

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public VsProjectProperty(string name, string value)
         {
             if (string.IsNullOrEmpty(name))
             {
-                throw new ArgumentException(ProjectManagement.Strings.Argument_Cannot_Be_Null_Or_Empty, nameof(name));
+                throw new ArgumentException("Argument cannot be null or empty", nameof(name));
```
Comment (alpaix) [inline test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs:18]: Is there a replacement for that string?

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectRestoreInfo.cs [hunk 4]
```diff
@@ -26,7 +26,7 @@ internal class VsProjectRestoreInfo : IVsProjectRestoreInfo
         {
             if (string.IsNullOrEmpty(baseIntermediatePath))
             {
-                throw new ArgumentException(ProjectManagement.Strings.Argument_Cannot_Be_Null_Or_Empty, nameof(baseIntermediatePath));
+                throw new ArgumentException("Argument cannot be null or empty", nameof(baseIntermediatePath));
```
Comment (alpaix) [inline test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectRestoreInfo.cs]: the same

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsReferenceItem.cs [hunk 5]
```diff
@@ -15,7 +15,7 @@ public VsReferenceItem(string name, IVsReferenceProperties properties)
         {
             if (string.IsNullOrEmpty(name))
             {
-                throw new ArgumentException(ProjectManagement.Strings.Argument_Cannot_Be_Null_Or_Empty, nameof(name));
+                throw new ArgumentException("Argument cannot be null or empty", nameof(name));
```
Comment (alpaix) [inline test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsReferenceItem.cs]: the same

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsReferenceProperty.cs [hunk 6]
```diff
@@ -15,7 +15,7 @@ public VsReferenceProperty(string name, string value)
         {
             if (string.IsNullOrEmpty(name))
             {
-                throw new ArgumentException(ProjectManagement.Strings.Argument_Cannot_Be_Null_Or_Empty, nameof(name));
+                throw new ArgumentException("Argument cannot be null or empty", nameof(name));
```
Comment (alpaix) [inline test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsReferenceProperty.cs]: the same

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsTargetFrameworkInfo.cs [hunk 7]
```diff
@@ -24,7 +24,7 @@ internal class VsTargetFrameworkInfo : IVsTargetFrameworkInfo
         {
             if (string.IsNullOrEmpty(targetFrameworkMoniker))
             {
-                throw new ArgumentException(ProjectManagement.Strings.Argument_Cannot_Be_Null_Or_Empty, nameof(targetFrameworkMoniker));
+                throw new ArgumentException("Argument cannot be null or empty", nameof(targetFrameworkMoniker));
```
Comment (alpaix) [inline test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsTargetFrameworkInfo.cs]: the same

File: test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs [hunk 8]
```diff
@@ -307,7 +306,7 @@ public class NuGetPackageManagerTests
                 }
 
                 Assert.NotNull(alreadyInstalledException);
-                Assert.Equal(string.Format(Strings.PackageAlreadyExistsInProject, packageIdentity, msBuildNuGetProjectSystem.ProjectName),
+                Assert.Equal(string.Format("Package '{0}' already exists in project '{1}'", packageIdentity, msBuildNuGetProjectSystem.ProjectName),
```
Comment (alpaix) [inline test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs]: this one, too

File: test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs [hunk 9]
```diff
@@ -2199,7 +2198,8 @@ public Task TestPacManGetUnrestoredPackagesByDependencyOrder()
                         msBuildNuGetProject, packageIdentity, resolutionContext, new TestNuGetProjectContext(), sourceRepositoryProvider.GetRepositories().First(), null, token));
 
                 var errorMessage = string.Format(CultureInfo.CurrentCulture,
-                    Strings.UnableToFindCompatibleItems, packageIdentity.Id + " " + packageIdentity.Version.ToNormalizedString(), msBuildNuGetProject.MSBuildNuGetProjectSystem.TargetFramework);
+                    "Could not install package '{0}'. You are trying to install this package into a project that targets '{1}', but the package does not contain any assembly references or content files that are compatible with that framework. For more information, contact the package author.",
```
Comment (alpaix) [inline test/NuGet.Core.Tests/NuGet.PackageManagement.Test/NuGetPackageManagerTests.cs]: Move to Strings

**Review by alpaix — CHANGES_REQUESTED** [review]
Address the feedback

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs (hunk 3 above)
Comment (nkolev92) [reply test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs:18]: Strings.Argument_Cannot_Be_Null_Or_Empty & Resources.Argument_Cannot_Be_Null_Or_Empty ?

**Review by nkolev92 — COMMENTED** [review]
After Alex's comments 🚀

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Strings.Designer.cs (hunk 1 above)
Comment (emgarten) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Strings.Designer.cs]: Using ``Run Custom Tool`` through VS automatically set this. Maybe the project needs to override the default namespace. I'll take a look.

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs (hunk 3 above)
Comment (emgarten) [reply test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs:18]: Yeah, it's the same string. I don't think we need to use localized strings in tests, let me know if there is a reason to.

File: src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs (hunk 2 above)
Comment (emgarten) [reply src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs]: Good catch

Comment (emgarten) [reply src/NuGet.Core/NuGet.PackageManagement/NuGetPackageManager.cs]: I've updated this

File: src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Strings.Designer.cs (hunk 1 above)
Comment (emgarten) [reply src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/Strings.Designer.cs]: This is updated

**Comment by emgarten** [comment]
🔔 addressed feedback

**Review by jainaashish — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs (hunk 3 above)
Comment (jainaashish) [reply test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs:18]: I'll leave it upto you to take a call. But i think It's more convenient to use resource strings even in tests then when string change we don't need do update all the tests again. Currently, I see lot of hard coded strings in our most tests to verify error/ warning messages.

Comment (alpaix) [reply test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs:18]: Ideally we should be using `Assumes.NotNullOrEmpty` or similar validation API.

**Review by alpaix — APPROVED** [review]

File: test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs (hunk 3 above)
Comment (jainaashish) [reply test/NuGet.Clients.Tests/NuGet.SolutionRestoreManager.Test/VsProjectProperty.cs:18]: or `Assumes.Present` which will throw appropriate argument null exception with right error message.

**Pull request merged**